### PR TITLE
explorer: support `copy` for dnd

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

Closes: #11870.
Closes: #9922.

The pull-request adds support to `copy` resources when we perform a drag-and-drop in the explorer with a modifier key present  (<kbd>ctrl</kbd> for Linux/Windows and <kbd>alt</kbd> for MacOS).

#### How to test

1. open a workspace with folders present
2. dnd files/folders to different locations in the explorer (a move icon should be present and a move should occur)
3. dnd files/folders to different locations in the explorer with a modifier key pressed (a copy icon should be present and a copy should occur)
4. the feature should work at different folder levels and with one or many resources

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/filesystem/src/browser/file-tree/file-tree-widget.tsx [hunk 1]
```diff
@@ -207,6 +213,11 @@ export class FileTreeWidget extends CompressedTreeWidget {
         return DirNode.getContainingDir(node);
     }
 
+    protected getDropEffect(event: React.DragEvent): 'copy' | 'move' {
+        const isCopy = (event.ctrlKey && !isOSX) || (event.altKey && isOSX);
```
Comment (colin-grant-work) [inline packages/filesystem/src/browser/file-tree/file-tree-widget.tsx:217]: I'd probably do something like 
```suggestion
        const isCopy = isOSX ? event.altKey : event.ctrlKey;
```

**Review by colin-grant-work — APPROVED** [review]
The behavior is now correct with respect to the copy modifier. As we discussed offline, the expansion of a folder on drag enter can result in jerky behavior, but that's unrelated to this change and could be handled in a separate PR. If you'd prefer to handle it here, ping me when it's ready :-).

**Comment by marcdumais-work** [comment]
Vince, can you add this issue as well, to be closed by your PR: #9922

**Review by marcdumais-work — APPROVED** [review]
LGTM, thanks Vince!

**Pull request merged**